tools/serverpod_cli/lib/src/generator/dart/library_generators/util/class_generators_util.dart

```diff
@@ -380,7 +380,12 @@ Expression _buildClassTypeFromJson(
           config: config,
         )
         .property('fromJson')
-        .call([valueExpression.asA(refer('Map<String, dynamic>'))])
+        .call([
+          if (!type.customClass)
+            valueExpression.asA(refer('Map<String, dynamic>'))
+          else
+            valueExpression,
+        ])
```

Comment:
Suggestion: Flip this if/else statement, that is remove the `!`.

That way the reader doesn't have to process the negation.